docs/book/src/tasks/experimental-features/cluster-class/write-clusterclass.md

```diff
@@ -1,3 +1,570 @@
+    * [Builtin variables](#builtin-variables)
+    * [JSON patch values](#json-patch-values)
+    * [Complex variables](#complex-variables)
+    * [Optional patches](#optional-patches)
+    * [MachineDeployment variable overrides](#machinedeployment-variable-overrides)
+
+## Basic ClusterClass
+
+The following example shows a basic ClusterClass. It contains templates to shape the control plane, 
+infrastructure and workers of a Cluster.  When a Cluster is using this ClusterClass, the templates 
+are used to generate the objects of the managed topology of the Cluster.
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: ClusterClass
+metadata:
+  name: docker-clusterclass-v0.1.0
+spec:
+  controlPlane:
+    ref:
+      apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+      kind: KubeadmControlPlaneTemplate
+      name: docker-clusterclass-v0.1.0
+      namespace: default
+    machineInfrastructure:
+      ref:
+        kind: DockerMachineTemplate
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        name: docker-clusterclass-v0.1.0
+        namespace: default
+  infrastructure:
+    ref:
+      apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+      kind: DockerClusterTemplate
+      name: docker-clusterclass-v0.1.0-control-plane
+      namespace: default
+  workers:
+    machineDeployments:
+    - class: default-worker
+      template:
+        bootstrap:
+          ref:
+            apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+            kind: KubeadmConfigTemplate
+            name: docker-clusterclass-v0.1.0-default-worker
+            namespace: default
+        infrastructure:
+          ref:
+            apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+            kind: DockerMachineTemplate
+            name: docker-clusterclass-v0.1.0-default-worker
+            namespace: default
+```
+
+The following example shows a Cluster using this ClusterClass. In this case a `KubeadmControlPlane` 
+with the corresponding `DockerMachineTemplate`, a `DockerCluster` and a `MachineDeployment` with 
+the corresponding `KubeadmConfigTemplate` and `DockerMachineTemplate` will be created. This basic 
+ClusterClass is already very flexible. Via the topology on the Cluster the following can be configured:
+* `.spec.topology.version`: the Kubernetes version of the Cluster
+* `.spec.topology.controlPlane`: control plane replicas and their metadata
+* `.spec.topology.workers`: MachineDeployments and their replicas, metadata and failure domain
+    * **Note**: It's possible to create MachineDeployments with different configurations, by
+      defining multiple MachineDeployment classes in the ClusterClass. 
+
+```yaml
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: my-docker-cluster
+spec:
+  topology:
+    class: docker-clusterclass
+    version: v1.22.4
+    controlPlane:
+      replicas: 3
+      metadata:
+        labels: {}
```

Comment:
It might be clearer to add  a sample label / annotation here to show usage - same below.

Reply:
My idea was to keep it short, but I added labels now. If you've got a good idea for labels/annotations please let me know :)